src/main/java/com/redhat/cloud/notifications/processors/camel/CamelTypeProcessor.java

```diff
@@ -0,0 +1,129 @@
+package com.redhat.cloud.notifications.processors.camel;
+
+import com.redhat.cloud.notifications.db.converters.MapConverter;
+import com.redhat.cloud.notifications.models.BasicAuthentication;
+import com.redhat.cloud.notifications.models.CamelProperties;
+import com.redhat.cloud.notifications.models.Endpoint;
+import com.redhat.cloud.notifications.models.Notification;
+import com.redhat.cloud.notifications.models.NotificationHistory;
+import com.redhat.cloud.notifications.processors.EndpointTypeProcessor;
+import com.redhat.cloud.notifications.transformers.BaseTransformer;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.opentelemetry.context.Context;
+import io.smallrye.mutiny.Uni;
+import io.smallrye.reactive.messaging.TracingMetadata;
+import io.vertx.core.json.JsonObject;
+import org.eclipse.microprofile.reactive.messaging.Channel;
+import org.eclipse.microprofile.reactive.messaging.Emitter;
+import org.eclipse.microprofile.reactive.messaging.Message;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.logging.Logger;
+
+import static com.redhat.cloud.notifications.models.NotificationHistory.getHistoryStub;
+
+
+@ApplicationScoped
+public class CamelTypeProcessor implements EndpointTypeProcessor {
+
+    private final Logger log = Logger.getLogger(this.getClass().getName());
+
+    private static final String TOKEN_HEADER = "X-Insight-Token";
+
+    @Inject
+    BaseTransformer transformer;
+
+    @Inject
+    @Channel("toCamel")
+    Emitter<String> emitter;
+
+    MeterRegistry registry;
+
+    private Counter processedCount;
+
+    public CamelTypeProcessor(MeterRegistry registry) {
+        this.registry = registry;
+        processedCount = registry.counter("processor.camel.processed");
```

Comment:
I've seen that elsewhere in the app and it seems to work, but considering `MeterRegistry` is a bean injected by Arc, it should not be used from a constructor in theory but rather from a `@PostConstruct`-annotated method.